statsmodels/regression/recursive_ls.py

```diff
@@ -267,6 +266,176 @@ def __init__(self, model, params, filter_results, cov_type='opg',
+                         coef    std err          t      P>|t|      [0.025      0.975]
+        ------------------------------------------------------------------------------
+        c0            15.0977     84.937      0.178      0.863    -177.042     207.238
+        c1            -2.0202      0.488     -8.231      0.000      -3.125      -0.915
+        c2             1.0001      0.249      0.000      1.000       0.437       1.563
+        ==============================================================================
+
+        See Also
+        ---------
+        tvalues : individual t statistics
+        f_test : for F tests
+        patsy.DesignInfo.linear_constraint
+        """
+        from statsmodels.tools.tools import recipr
+        from statsmodels.stats.contrast import ContrastResults
+        from patsy import DesignInfo
+        names = self.model.data.param_names
+        LC = DesignInfo(names).linear_constraint(r_matrix)
+        r_matrix, q_matrix = LC.coefs, LC.constants
+        num_ttests = r_matrix.shape[0]
+        num_params = r_matrix.shape[1]
+
+        # Check we don't have conflicting `loc` and `cov_p`
+        if loc is not None and cov_p is not None:
+            raise ValueError('Cannot use `loc` in combination with `cov_p`.')
+
+        if num_params != self.params.shape[0]:
+            raise ValueError('r_matrix and params are not aligned')
+        if q_matrix is None:
+            q_matrix = np.zeros(num_ttests)
+        else:
+            q_matrix = np.asarray(q_matrix)
+            q_matrix = q_matrix.squeeze()
+        if q_matrix.size > 1:
+            if q_matrix.shape[0] != num_ttests:
+                raise ValueError("r_matrix and q_matrix must have the same "
+                                 "number of rows")
+
+        if use_t is None:
+            # switch to use_t false if undefined
+            use_t = (hasattr(self, 'use_t') and self.use_t)
+
+        _t = _sd = None
+
+        # If we've been given a location
+        if loc is not None:
+            t, _, out_of_sample, _ = (
+                self.model._get_prediction_index(loc, loc, index=None))
+            if t > self.model.nobs:
+                raise ValueError('Given t-test location is not in the sample.')
+            # Get the appropriate cov
+            params = self.filtered_state[:, t]
+            cov_p = self.filtered_state_cov[:, :, t]
+
+            # Rescale the covariance parameters based on the scale estiamted
+            # as of observation t
+            d = max(self.loglikelihood_burn, self.nobs_diffuse)
+            nmissing = np.sum(self.filter_results.nmissing[d:t + 1])
+            n = (t + 1 - d) - nmissing
+            scale = (np.sum(self.filter_results.scale_obs[d:t + 1]) / n)
+            cov_p *= scale / self.scale
+        else:
+            d = max(self.loglikelihood_burn, self.nobs_diffuse)
+            t = self.model.nobs
```

Comment:
is `t` used?

Reply:
Down below in the `nobs_effective` computation.